### PR TITLE
Feat/doc sign

### PR DESCRIPTION
**Pull request by jesusantguerrero**
- Add support for fixed filters in gw-browse
- Add support for date range
- Add top left pagination slots
- Add actions slot

### CODE REVIEW

File: src/browse.vue [hunk 1]
```diff
@@ -19,49 +19,56 @@
...
-            :filterable-fields="filterableFields"
-            :search-options="searchOptions"
-            :show-bulk-actions="showBulkActions"
-            :show-create-resource="showCreateResource"
-            :show-search-filters="showSearchFilters"
-            @getData="getData"
-            @run-action="runAction"
-            @show-custom-filters-form="$modal.show('custom-filters-form')"
-        />
+        <slot :data="{searchOptions}" name="resource-actions" v-bind="{ searchOptions, getData, filterableFields }">
```
Comment (SparoHawk) [inline src/browse.vue:22]:
```suggestion
        <slot :data="{ searchOptions }" name="resource-actions" v-bind="{ searchOptions, getData, filterableFields }">
```

File: src/browse.vue [hunk 2]
```diff
@@ -19,49 +19,56 @@
...
-                            |
-                        </div>
-                    </template>
-                    <vuetable-pagination
-                        ref="paginationTop"
-                        :css="pagination"
-                        class="col-auto"
-                        @vuetable-pagination:change-page="onChangePage"
-                    />
+                    <slot name="before-table-left" />
```
Comment (SparoHawk) [inline src/browse.vue:43]:
```suggestion
                    <slot name="before-pagination" />
```

File: src/browse.vue [hunk 3]
```diff
@@ -370,13 +377,19 @@ export default {
...
+            return mainDateField.length ? mainDateField[0] : "";
         }
     },
     watch: {
         appendParams() {
             this.vuetableQueryParams = _clone(this.appendParams);
         },
         resource() {
+            this.$refs.Vuetable.resetData()
+            this.vuetableQueryParams.q = null
```
Comment (SparoHawk) [inline src/browse.vue:392]:
```suggestion
            this.$refs.Vuetable.resetData();
            this.vuetableQueryParams.q = null;
```

File: src/browse.vue [hunk 4]
```diff
@@ -440,16 +453,22 @@ export default {
         },
         getData(searchOptions) {
             let params = "";
+            const fixedFilters = Object.keys(searchOptions.fixedFilters || {})
+            const dateFilters = Object.keys(searchOptions.dates || {})
```
Comment (SparoHawk) [inline src/browse.vue:457]:
```suggestion
            const fixedFilters = Object.keys(searchOptions.fixedFilters || {});
            const dateFilters = Object.keys(searchOptions.dates || {});
```

File: src/browse.vue [hunk 5]
```diff
@@ -440,16 +453,22 @@ export default {
...
+            let searchableFields = [];
             searchOptions.text = searchOptions.text.trim();
 
             if (searchOptions.text.length) {
                 if (!searchOptions.filters.length) {
-                    params += this.getParams(this.searchableFields, searchOptions);
+                    searchableFields = this.searchableFields.filter(field => !fixedFilters.includes(field) && !dateFilters.includes(field))
```
Comment (SparoHawk) [inline src/browse.vue:463]:
```suggestion
                    searchableFields = this.searchableFields.filter(field => !fixedFilters.includes(field) && !dateFilters.includes(field));
```

File: src/browse.vue [hunk 6]
```diff
@@ -440,16 +453,22 @@ export default {
...
                 } else {
-                    params += this.getParams(searchOptions.filters, searchOptions);
+                    searchableFields = searchOptions.filters.filter(field => !fixedFilters.includes(field) && !dateFilters.includes(field))
```
Comment (SparoHawk) [inline src/browse.vue:465]:
```suggestion
                    searchableFields = searchOptions.filters.filter(field => !fixedFilters.includes(field) && !dateFilters.includes(field));
```

File: src/browse.vue [hunk 7]
```diff
@@ -440,16 +453,22 @@ export default {
...
                 }
+                params += this.getParams(searchableFields, searchOptions);
             }
 
+            params = this.getFixedFilters(searchOptions, params)
```
Comment (SparoHawk) [inline src/browse.vue:470]:
```suggestion
            params = this.getFixedFilters(searchOptions, params);
```

File: src/browse.vue [hunk 8]
```diff
@@ -469,6 +488,32 @@ export default {
                 (this.showActionsDelete || this.showActionsEdit) && this.tableFields.push(this.vuetableActions);
             });
         },
+
+        getFixedFilters(searchOptions, params) {
+            let fixedFilters = Object.entries(searchOptions.fixedFilters || {});
+            const dateFilters = Object.entries(searchOptions.dates || {});
+            let dateValues = ""
```
Comment (SparoHawk) [inline src/browse.vue:495]:
```suggestion
            let dateValues = "";
```

File: src/browse.vue [hunk 9]
```diff
@@ -469,6 +488,32 @@ export default {
...
+
+            if (fixedFilters.length || dateFilters.length) {
+                fixedFilters = fixedFilters.map(([filterName, value]) => `${filterName}:${value}`).join(";");
+
```
Comment (SparoHawk) [inline src/browse.vue:499]:
```suggestion

```

File: src/browse.vue [hunk 10]
```diff
@@ -469,6 +488,32 @@ export default {
...
+                        dateFilterValue += `,${this.mainDateField}<${this.formatDate(searchOptions.dates.end)}`;
+                    }
+                    dateValues = `${this.mainDateField}>${dateFilterValue}`;
+                }
+                params = [params, dateValues, fixedFilters].filter(val => val).join(",");
+            }
+
+            return params;
+        },
+
```
Comment (SparoHawk) [inline src/browse.vue:512]:
```suggestion
```

File: src/browse.vue [hunk 11]
```diff
@@ -469,6 +488,32 @@ export default {
...
+        formatDate(date) {
+            return date ? date.toISOString().slice(0, 10) : ""
```
Comment (SparoHawk) [inline src/browse.vue:514]:
```suggestion
            return date ? date.toISOString().slice(0, 10) : "";
```

File: src/browse.vue [hunk 12]
```diff
@@ -469,6 +488,32 @@ export default {
...
+        },
+
```
Comment (SparoHawk) [inline src/browse.vue:516]:
```suggestion
```

File: src/browse.vue [hunk 13]
```diff
@@ -370,13 +377,19 @@ export default {
         },
         searchableFields() {
             return this.tableFields.filter(field => field.searchable).map(field => field.name);
+        },
+        mainDateField() {
```
Comment (SparoHawk) [inline src/browse.vue:381]: So what happens where there is more than one date field in the filterable fields?

Comment (jesusantguerrero) [reply src/browse.vue:381]: It should't be more than one by the moment that's why is called mainDateField; but later, maybe we can allow the User decide which `date field` to use or address the problem in custom filters. 

what do you think?

Comment (SparoHawk) [reply src/browse.vue:381]: Later is okay.

**Pull request merged**